### PR TITLE
Add progress indicator to loading screen

### LINKED ISSUE #691: Analyzation progress indicator

### PR DESCRIPTION
**Pull request by tjomson**
Resolves #691

### CODE REVIEW

**Review by joglr — COMMENTED** [review]
Very cool feature!

Could we adjust the text tracking a bit so it is not as wide?

**Review by joglr — APPROVED** [review]
Awesome!

File: src/components/LoadingIndicator.tsx [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+    const intervalId = setInterval(() => {
+      if (fetcher.state === "idle") fetcher.load(`/progress`)
+    }, 3000)
+
+    return () => clearInterval(intervalId)
+  }, [])
+
+  const progressText = useMemo(() => {
+    if (!fetcher.data) return "Starting analyzation"
+    const { progress, totalCommitCount, analyzationStatus } = fetcher.data as {
```
[the remark this answers is not in the document]
Comment (tjomson) [reply src/components/LoadingIndicator.tsx:17]: Good point :O

File: src/components/LoadingIndicator.tsx [hunk 2]
```diff
@@ -1,13 +1,31 @@
...
 import anitruck from "~/assets/truck.gif"
 
-export function LoadingIndicator({
-  loadingText = "Loading",
-  className = ""
-}: {
-  loadingText?: string
-  className?: string
-}) {
+interface ProgressData {
```
Comment (joglr) [inline src/components/LoadingIndicator.tsx:7]: Prefer type over interfaces

https://www.totaltypescript.com/type-vs-interface-which-should-you-use

**Review by joglr — CHANGES_REQUESTED** [review]

**Review by joglr — APPROVED** [review]

**Pull request merged**